docs/tech_primer.md

```diff
@@ -0,0 +1,157 @@
+# Smart-Ready Building Technical Primer
+
+This document provides  primer for smart-ready building assembly, to essentially
+answer the question *"What does Smart Ready mean?"* and *"How do we know it to be
+true?"* Specifically, it addresses the delta between a traditional site and
+smart-ready sites: places where new bleeding edge building technology is being
+used.
+
+# Key Terms & Definitions
+
+## Building Models
+
+There are different levels of "digital maturity" for different buildings. Each
+one has a slightly different scope in terms of what it means to be "complete."
+For the most part, this document addresses the scope of a Smart Ready building:
+-   **Legacy**: It works, but does not meet requirements or guidelines.
+-   **Compliant**: Meets basic security and networking requirements.
+-   **Smart Ready**: On-prem integrated with the cloud using UDMI & DBO.
+-   **Digital Building**: Completely integrated with back-end services.
+
+## Device
+
+Hardware that exists in the building that can change state in a way that can be machine communicated. A potentiometer or a flow switch is a device if it is monitored by a digital system. A commissioning valve or damper with no communication is not a device. 
+
+## Devices Classification
+
+Various adjectives qualify the different flavors of a "device" found in a building. More than one can apply, and each has a specific technical meaning/definition (i.e. they are not subjective).
+
+### Connectivity
+
+How data is moved around on-prem
+
+-   **Passive**: The device has only passive components (thermistor, dry contact)
+-   **Analog**: The device uses an analog connection (1-10V, 4-20mA etc)
+-   **Serial**: The device does not have an IP address, rather uses a wired serial connection.
+-   **Networked**: The device has its own IP address and presence on the wired or wireless IP network.
+-   **Bridge connected**: The device is on an isolated wired or wireless network behind a gateway device that is networked. 
+
+### Ingestion
+
+How data gets up into the cloud
+
+-   **Direct**: Maintains an authenticated connection directly to Cloud IoT Core.
+    -   Must be networked
+    -   Has a unique private auth key
+-   **Gateway**: A direct device that manages data for other proxied devices.
+    -   Bind & attach of proxied devices
+    -   Encapsulation anti-pattern
+-   **Proxied**: A device that is not direct and is managed through a gateway.
+    -   Proxied networked devices, e.g. BACnet/IP
+    -   Proxied non-networked devices, e.g. RS485, DALI
+-  **Cloud-Cloud**: A device is managed by a third party who provides a cloud connector
+    -   Approved third party managed service
```

Comment:
"third party" and "managed" are not _technical_ concepts so shouldn't be here. What is _technically_ different about this that needs to be covered?

Reply:
The idea here is a device installed in a building which communicates over some proprietary protocol to a SaaS platform. This platform has an API to get data from the device. A connector may be written which reads/interacts from this API and publishes data over MQTT/UDMI. The connector may be running as a virtual machine somewhere. This means some of the physical aspects may not be applicable. I think the desire is UDMI is supported from devices, so this use case may be an exception and can be removed from this document

Reply:
Additional note - this may apply to VT/lift suppliers in future projects, or some IoT sensor products.

Reply:
So separate out the technical aspects (should be included) of what you said, with the administrative ones (should not be included). "Third party" is administrative, not technical.  I think what you're describing is something like "external" -- which is indicating that the data source is _external_ to the "confined" system. Or something like that that characterizes the architecture that you outline.

Reply:
I used the words `**External** A device (on-prem or off-prem) consumes data from an external source and publishes it to Cloud IoT core`